rule-audit-client/src/redux/modules/searchMatches/reducer.ts

```diff
@@ -0,0 +1,31 @@
+import { createReducer } from "typesafe-actions";
+import { doSetSearchMatchesLimit, doSearchMatchesStart, doAppendSearchMatchesArticleIds, doSearchMatchesEnd } from "./actions";
+
+const initialState = {
+  searchMatchesLimit: 10,
+  isSearchMatchesInProgress: false,
+  searchMatchesArticleIds: [] as string[]
+};
+
+export type SearchMatchesState = typeof initialState
+
+const reducer = createReducer(initialState)
+  .handleAction(doSetSearchMatchesLimit, (state, action) => ({
+    ...state,
+    searchMatchesLimit: action.payload,
+  }))
```

Comment:
I don't think I've seen "typesafe-actions" `createReducer` used before. This is so much more concise and readable than the usual reducer structure! 👍